components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/ClientAuthX509TrustManager.java

```diff
@@ -0,0 +1,136 @@
+
+/**
+ * Gives a javax.net.ssl.X509TrustManager implementation that uses the default carbon trust store which can be used
+ * by tomcat connector to dynamically reload the client truststore whenever truststore content changes.
+ * This will load any changes (addition/removal of certificates) done to the default trust store on the fly.
+ */
+public class ClientAuthX509TrustManager implements X509TrustManager {
+
+    private static Log log = LogFactory.getLog(ClientAuthX509TrustManager.class);
+    private X509TrustManager trustManager;
+
+    //Configuration Options
+    private static final ServerConfiguration config = ServerConfiguration.getInstance();
+    private static final String TRUSTSTORE_LOCATION = config.getFirstProperty("Security.TrustStore.Location");
+    private static final String TRUSTSTORE_TYPE = config.getFirstProperty("Security.TrustStore.Type");
+
+    public ClientAuthX509TrustManager() throws Exception {
+
+        setupTrustManager();
+    }
+
+    @Override
+    public void checkClientTrusted(X509Certificate[] x509Certificates, String s) throws CertificateException {
+
+        try {
+            //if changes were made to the trust store, reload the trust store and initialize the trustManager instance.
+            if ("true".equals(System.getProperty(PROP_TRUST_STORE_UPDATE_REQUIRED))) {
+                setupTrustManager();
+            }
+            trustManager.checkClientTrusted(x509Certificates, s);
+        } catch (NoSuchAlgorithmException | KeyStoreException e) {
+            throw new CertificateException("Certificate validation failed due to " + e.getCause(), e);
+        } catch (Exception e) {
+
+            throw new CertificateException("Certificate validation failed due to " + e.getCause(), e);
+        }
+    }
+
+    /**
+     * Checks the validity of passed x509Certificate certificate chain
+     *
+     * @param x509Certificates
+     * @param s
+     * @throws CertificateException
+     */
+    @Override
+    public void checkServerTrusted(X509Certificate[] x509Certificates, String s) throws CertificateException {
+
+        trustManager.checkServerTrusted(x509Certificates, s);
+
+    }
+
+    @Override
+    public X509Certificate[] getAcceptedIssuers() {
+
+        try {
+            // Reload the trust store and initialize the trustManager instance for every SSL session.
+                setupTrustManager();
+        } catch (Exception e) {
+            log.error("Error occurred while reloading trust-store.");
+        }
+        return trustManager.getAcceptedIssuers();
+    }
+
+
+    /**
+     * This method reloads the TrustManager by reading the carbon server's default trust store file
+     *
+     * @throws Exception
+     */
+    private void setupTrustManager() throws Exception {
+
+        TrustManagerFactory trustManagerFactory =
+                TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
```

Comment:
Let's check the impact of setting the `TrustManagerFactory.getDefaultAlgorithm()`